rclcpp/test/rclcpp/test_timers_manager.cpp

```diff
@@ -397,11 +398,30 @@ TEST_F(TestTimersManager, check_one_timer_cancel_doesnt_affect_other_timers)
   // Start timers thread
   timers_manager->start();
 
-  std::this_thread::sleep_for(15ms);
+  // Wait for t1 to be canceled
+  while (!t1->is_canceled()) {
```

Comment:
Can we put an overall timeout on this, something like 30 seconds?  It just ensures that if there is a bug, this test won't hang our CI forever.

Reply:
Done in e06bd91a8979b6218fd8bca530bf38750dec1524